File: frontend/src/api/taskService.js
import api from './axiosConfig';

export const getTasks = async (taskListId) => {
    const response = await api.get(`/task-lists/${taskListId}/tasks`);
    return response.data;
};

export const createTask = async (taskListId, task) => {
    const response = await api.post(`/task-lists/${taskListId}/tasks`, task);
    return response.data;
};

export const updateTask = async (taskListId, taskId, task) => {
    const response = await api.put(`/task-lists/${taskListId}/tasks/${taskId}`, task);
    return response.data;
};

export const deleteTask = async (taskListId, taskId) => {
    await api.delete(`/task-lists/${taskListId}/tasks/${taskId}`);
};

File: src/main/java/com/example/taskaz/services/TaskService.java
package com.example.taskaz.services;

import com.example.taskaz.domain.dto.TaskDto;
import com.example.taskaz.domain.entity.Task;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface TaskService {
    List<Task> listTasks(UUID taskListId);
    Task createTask(UUID taskListId, Task task);
    Optional<Task> getTask(UUID id,UUID taskListId);
    Task updateTask(UUID taskListId, UUID taskId, Task task);
    void deleteTask(UUID taskListId, UUID taskId);
}
